infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/endpoints/EndpointMetadata.java

```diff
@@ -103,6 +106,31 @@ public static EndpointMetaDataBuilder delete(final String path) {
     return new EndpointMetaDataBuilder().method(HandlerType.DELETE).path(path);
   }
 
+  @SuppressWarnings({"unchecked", "TypeParameterUnusedInFormals"})
+  public <T> T getRequestBody(final String body) throws JsonProcessingException {
+    checkArgument(requestBodyType.isPresent(), "requestBodyType has not been defined");
+    checkArgument(
+        requestBodyTypeSelector.isPresent(), "requestBodyTypeSelector has not been defined");
+
+    final DeserializableTypeDefinition<T> bodySchema =
+        (DeserializableTypeDefinition<T>) getRequestBodyType(body);
+
+    if (bodySchema == null) {
+      throw new MissingRequestBodyException();
+    }
+
+    if (!getRequestBodyType().getSelfAndReferencedTypeDefinitions().contains(bodySchema)) {
+      throw new IllegalStateException(
+          "schema determined for parsing request body is not listed in requestBodyTypes");
+    }
```

Comment:
this is not 100% accurate, not sure if there's a better way of encapsulating oneOf AND normal types?
I can forsee this would allow us to send back UInt64 because its included somewhere deep...

Reply:
Yeah definitely not what we want but right idea.  I think we need a new method on `OpenApiTypeDefinition` which is a bit hard to name but something like `isEquivalentDeserializableType(DeserializableTypeDefinition<?> type)`.  `OpenApiTypeDefinition` would have a default implementation which just checks that `type == this` (ie we're using the same type so it must be equivalent).

Then `SerializableOneOfTypeDefinition` would override that and for each of it's options call `option.isEquivalentDeserializableType(type)` and if any of them match return true.

I think in this case I'd actually be keen to use directly equality comparison as well - it's very difficult to say if two different instances are actual equivalent serialisations and we want to avoid creating new instances when we don't have to so I like the idea of having to use the same instance to be considered equivalent.  It will probably mean we need to suppress error prone warnings about using `==` and will need a comment explaining it but I think it's the right thing in this particular case.